.travis.yml

```diff
@@ -17,18 +17,65 @@ sudo: false
 language: java
 jdk:
   - openjdk11
-before_install:
-  - echo 'MAVEN_OPTS="-Xmx4096M -Xss128M -XX:+CMSClassUnloadingEnabled -XX:+TieredCompilation -XX:TieredStopAtLevel=1"' >~/.mavenrc
-# default install is mvn install --quiet -DskipTests=true
-install: travis_wait 30 mvn --show-version --quiet -T 1C -P all,skipTests
-before_script:
-  - cd fit/core-reference
-script:
-  # invoker.streamLogs: we cannot access to log files through Travis web ui, so display everything in the console
-  - mvn --show-version verify -Dinvoker.streamLogs=true
-after_failure:
-  - cat target/log/*
-  - cat target/failsafe-reports/org.apache.syncope.fit.*-output.txt
+cache:
+  bundler: false
+  cargo: false
+  directories:
+    - $HOME/.m2
+git:
+  depth: 2
+matrix:
+  fast_finish: true
+env:
+  global:
+  - MAVEN_OPTS="-Xmx4096M -Xss128M -XX:+CMSClassUnloadingEnabled -XX:+TieredCompilation -XX:TieredStopAtLevel=1 -noverify"
+install: true
 notifications:
   email:
     - dev@syncope.apache.org
+# after_failure:
+#   - cat target/log/*
+#   - cat target/failsafe-reports/org.apache.syncope.fit.*-output.txt
+# before_script:
+#   -
+#   -
+# script:
+stages:
+  - initialize
+  - build
+  - validate
+  - test
+  - finalize
+jobs:
+  include:
+    - stage: initialize
+      script: mvn --quiet --show-version -T 4 enforcer:enforce
+      name: "Verify Platform"
+    - stage: initialize
+      script: mvn -T 4 clean dependency:go-offline dependency:resolve-plugins --quiet -Denforcer.skip=true
+      name: "Warm Up Dependencies"
+      #######################################################
+    - stage: build
+      script: mvn install --quiet -T 4 -P all,skipTests,skipLegalFiles -Dcheckstyle.skip=true -Dmaven.javadoc.skip=true -Denforcer.skip=true
+      name: "Build"
+      #######################################################
+    - stage: validate
+      script: mvn checkstyle:check -T 4 -Denforcer.skip=true --quiet
+      name: "Checkstyle Analysis"
+    - stage: validate
+      script: mvn -T 4 --quiet javadoc:javadoc javadoc:test-javadoc -Denforcer.skip=true
+      name: "Javadoc Analysis"
+    - stage: validate
+      script: mvn -T 4 --quiet ianal:verify-legal-files -Denforcer.skip=true
+      name: "Legal Files Analysis"
+    - stage: validate
+      script: mvn -T 4 --quiet modernizer:modernizer
+      name: "Modernizer Legacy API Analysis"
+      ######################################################
+    - stage: test
+      script: mvn -f fit/core-reference/pom.xml verify --quiet -T 4 -Dinvoker.streamLogs=true -Denforcer.skip=true -Dmodernizer.skip=true -Djacoco.skip=true
```

Comment:
I have seen the usual error in https://travis-ci.org/apache/syncope/jobs/583783048

```
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received
```

Maybe you can change this line to

````
travis_wait 30 mvn -f fit/core-reference/pom.xml verify -T 4 -Dinvoker.streamLogs=true -Denforcer.skip=true -Dmodernizer.skip=true -Djacoco.skip=true
```

e.g. remove `--quiet` and raise the silence timeout to 30 minutes instead of 10.

Reply:
This will likely do the job, but does have the drawback that all build output is basically silenced (to whatever degree we have them on) because the job is basically run in the background.

I'll wait to see if the build passes, but this strategy likely needs to be reformulated to be something like this instead:

```
waitloop="while sleep 9m; do echo -e '\n=====[build is still running ]====='; done &"
eval $waitloop
waitRetVal=$?
```

And then we execute the maven task. This is basically the same thing as `travis_wait` except output is kept alive from the build.